homeassistant/components/zha/radio_manager.py

```diff
@@ -156,3 +205,138 @@ async def async_reset_adapter(self) -> None:
         """Reset the current adapter."""
         async with self._connect_zigpy_app() as app:
             await app.reset_network_info()
+
+    async def async_restore_backup_step_1(self) -> bool:
+        """Prepare restoring backup.
+
+        Returns True if async_restore_backup_step_2 should be called.
+        """
+        assert self.chosen_backup is not None
+
+        if self.radio_type != RadioType.ezsp:
+            await self.restore_backup(self.chosen_backup)
+            return False
+
+        # We have no way to partially load network settings if no network is formed
+        if self.current_settings is None:
+            # Since we are going to be restoring the backup anyways, write it to the
+            # radio without overwriting the IEEE but don't take a backup with these
+            # temporary settings
+            temp_backup = _prevent_overwrite_ezsp_ieee(self.chosen_backup)
+            await self.restore_backup(temp_backup, create_new=False)
+            await self.async_load_network_settings()
+
+            assert self.current_settings is not None
+
+        if (
+            self.current_settings.node_info.ieee == self.chosen_backup.node_info.ieee
+            or not self.current_settings.network_info.metadata["ezsp"][
+                "can_write_custom_eui64"
+            ]
+        ):
+            # No point in prompting the user if the backup doesn't have a new IEEE
+            # address or if there is no way to overwrite the IEEE address a second time
+            await self.restore_backup(self.chosen_backup)
+
+            return False
+
+        return True
+
+    async def async_restore_backup_step_2(self, overwrite_ieee: bool) -> None:
+        """Restore backup and optionally overwrite IEEE."""
+        assert self.chosen_backup is not None
+
+        backup = self.chosen_backup
+
+        if overwrite_ieee:
+            backup = _allow_overwrite_ezsp_ieee(backup)
+
+        # If the user declined to overwrite the IEEE *and* we wrote the backup to
+        # their empty radio above, restoring it again would be redundant.
+        await self.restore_backup(backup)
+
+
+class ZhaMigrationHelper:
+    """Helper class for automatic migration."""
+
+    def __init__(
+        self, hass: HomeAssistant, config_entry: config_entries.ConfigEntry
+    ) -> None:
+        """Initialize MigrationHelper instance."""
+        self._config_entry = config_entry
+        self._hass = hass
+        self._radio_mgr = ZhaRadioManager()
+        self._radio_mgr.hass = hass
+
+    async def async_prepare_yellow_migration(self, data: dict[str, Any]) -> bool:
```

Comment:
Is this only for the yellow? It's not for users running SkyConnect eg?

Reply:
Yeah, it's intended to be used for both yellow and sky connect. I'll try to come up with a better name.

Reply:
Maybe make the name multi pan specific until there is another use case?

Reply:
OK, renamed again.